### PR TITLE
Hdinsight storm windows

### PR DESCRIPTION
**Pull request by morsh**
### Best Practice Checklist
Check these items before submitting a PR... See the Contribution Guide for the full detail: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md 

1. uri's compatible with all clouds (Stack, China, Government)
1. Staged artifacts use _artifactsLocation & _artifactsLocationSasToken
1. Use resourceGroup().location for resource locations
1. Folder names for artifacts (nestedtemplates, scripts, DSC)
1. Use literal values for apiVersion (no variables)
1. Parameter files (GEN-UNIQUE for value generation and no "changemeplease" values
1. $schema and other uris use https
1. Use uniqueString() whenever possible to generate names for resources.  While this is not required, it's one of the most common failure points in a deployment. 
1. Update the metadata.json with the current date

For details: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/bp-checklist.md

- [X ] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

### Changelog

Adding a template that contains:
* HDInsight on Windows with Storm
* Storage account used as the physical (WASB) storage for HDI
* IotHub used for stream analytics with storm

Adding this mainly  because there are almost no exmaples in 101 templates or the web on how to deploy an HDI on Azure that is based on Windows. (Let alone with storm)

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@morsh__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (morshe). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

**Comment by morsh** [comment]
NOTICE: I tested the template with a valid parameters file, but the CI requires real values instead of GEN-UNIQUE.

File: 301-azure-relay-create-authrule-namespace-and-hybridconnection/README.md [hunk 1]
```diff
@@ -1,7 +1,12 @@
 # Create an Azure Relay namespace & it's entities with SAS Policies
 
-<a href="https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fazure-quickstart-templates%2Fmaster%2F301-azurerelay-create-authrule-namespace-and-hybridconnection%2Fazuredeploy.json" target="_blank">
+<a href="https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fazure-quickstart-templates%2Fmaster%2F301-hdinsight-windows-storm-iothub%2Fazuredeploy.json" target="_blank">
```
Comment (gatneil) [inline 301-azure-relay-create-authrule-namespace-and-hybridconnection/README.md]: why are we pointing the README for 301-azure-relay-... to the template for 301-hdinsight-windows-...?

File: 301-azure-relay-create-authrule-namespace-and-hybridconnection/README.md [hunk 2]
```diff
@@ -1,7 +1,12 @@
...
 </a>
+<a href="http://armviz.io/#/?load=https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fazure-quickstart-templates%2Fmaster%2F301-hdinsight-windows-storm-iothub%2Fazuredeploy.json" target="_blank">
+    <img src="http://armviz.io/visualizebutton.png"/>
+</a>
+
+This template creates an HDInsight cluster based on windows with Storm next to an IotHub with Azure Storage for the HDInsight.
 
-This template creates an Azure Relay namespace, a HybridConnection and authorization rules for both the namespace and HybridConnection.
\ No newline at end of file
+This solution is meant as a base for storm based analytics using IotHub.
```
Comment (gatneil) [inline 301-azure-relay-create-authrule-namespace-and-hybridconnection/README.md]: I'm guessing the changes to this README were made to the wrong folder?

File: 301-hdinsight-windows-storm-iothub/README.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
+# Install Hue as an HDInsight Iaas Cluster Application
+
+Installs Hue as an Iaas Cluster Application on an existing cluster -<br>
+<a href="https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fazure-quickstart-templates%2Fmaster%2Fhdinsight-linux-hue-existing-cluster%2Fazuredeploy.json" target="_blank">
```
Comment (gatneil) [inline 301-hdinsight-windows-storm-iothub/README.md]: This README doesn't seem to match the folder name (neither the description nor the link)

File: 301-hdinsight-windows-storm-iothub/azuredeploy.json [hunk 4]
```diff
@@ -0,0 +1,267 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+
```
Comment (gatneil) [inline 301-hdinsight-windows-storm-iothub/azuredeploy.json]: please remove extra space

File: 301-hdinsight-windows-storm-iothub/azuredeploy.json [hunk 5]
```diff
@@ -0,0 +1,267 @@
...
+      "properties": {
+        "accountType": "[parameters('storageAccountType')]"
+      }
+    },
+
+    {
+      "name": "[parameters('stormClusterName')]",
+      "type": "Microsoft.HDInsight/clusters",
+      "location": "[variables('location')]",
+      "apiVersion": "2015-03-01-preview",
```
Comment (gatneil) [inline 301-hdinsight-windows-storm-iothub/azuredeploy.json]: is there a non-preview api version we can use?

File: 301-hdinsight-windows-storm-iothub/azuredeploy.json [hunk 6]
```diff
@@ -0,0 +1,267 @@
...
+            }
+          }
+        },
+        "storageProfile": {
+          "storageaccounts": [
+            {
+              "name": "[concat(parameters('storageAccountName'),'.blob.core.windows.net')]",
+              "isDefault": true,
+              "container": "[parameters('stormClusterName')]",
+              "key": "[listKeys(resourceId('Microsoft.Storage/storageAccounts', parameters('storageAccountName')), '2015-05-01-preview').key1]"
```
Comment (gatneil) [inline 301-hdinsight-windows-storm-iothub/azuredeploy.json]: non-preview api version?

File: 301-hdinsight-windows-storm-iothub/azuredeploy.json [hunk 7]
```diff
@@ -0,0 +1,267 @@
...
+              "restAuthCredential.isEnabled": true,
+              "restAuthCredential.username": "[parameters('stormClusterLoginUserName')]",
+              "restAuthCredential.password": "[parameters('stormClusterLoginPassword')]"
+            }
+          }
+        },
+        "storageProfile": {
+          "storageaccounts": [
+            {
+              "name": "[concat(parameters('storageAccountName'),'.blob.core.windows.net')]",
```
Comment (gatneil) [inline 301-hdinsight-windows-storm-iothub/azuredeploy.json]: Please see best practice 1 in the bp-checklist: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/bp-checklist.md

**Comment by gatneil** [comment]
Please address the above comments

File: 301-hdinsight-windows-storm-iothub/azuredeploy.json (hunk 5 above)
Comment (morsh) [inline 301-hdinsight-windows-storm-iothub/azuredeploy.json]: @gatneil -
All examples in this repo with hdinsight uses 2015-03-01-preview.
When I try a different version I get:
`
{
  "error": {
    "code": "NoRegisteredProviderFound",
    "message": "No registered resource provider found for location 'westeurope' and API version '2016-01-01' for type 'clusters'. The supported api-versions are '2015-03-01-preview'. The supported locations are 'eastasia, southeastasia, eastus, eastus2, westus, northcentralus, southcentralus, centralus, northeurope, westeurope, japaneast, japanwest, brazilsouth, australiaeast, australiasoutheast, centralindia'."
  }
}
`

**Comment by morsh** [comment]
Hi @gatneil,
Added the changes you requested. LMK if anything else is needed. Thanks!

**Comment by morsh** [comment]
Hi @gatneil, anything new? :)

File: 301-hdinsight-windows-storm-iothub/azuredeploy.json [hunk 8]
```diff
@@ -0,0 +1,266 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "storageAccountName": {
```
Comment (gatneil) [inline 301-hdinsight-windows-storm-iothub/azuredeploy.json]: Please see best practice 6: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md

**Comment by gatneil** [comment]
Sorry, I missed one thing; please address the above comment; otherwise looks good :)

**Comment by morsh** [comment]
@gatneil - added the changes you requested

**Comment by morsh** [comment]
@gatneil - small ping..

**Comment by bmoore-msft** [comment]
Looks like this might be too dated - if you do want to get it working, feel free to re-open or submit a new pr

**Pull request closed without merging**